# Entendiento el problema de negocio y su contexto

In [12]:
# import libraries
import pandas as pd
from optimus import Optimus
op = Optimus()

In [13]:
# Read the data
# Data from http://rpubs.com/rhuebner/HRCodebook-13
df = op.read.csv("data/hr-data.csv", header=True)

In [14]:
# See the data
df.table()

![](imgs/1.png)

In [17]:
df.describe().table()

# ¿Cuánto nos cuesta perder a un empleado?

In [32]:
# Calculate avg salary
avg_salary = df.agg({"PayRate":"avg"}).first()[0]

In [33]:
avg_salary_per_year = avg_salary * 2080

In [34]:
avg_salary_per_year

65072.39741935487

In [48]:
benefits = 14928

In [51]:
salary = round(avg_salary_per_year + benefits,0)
salary

80000.0

In [53]:
# Costos directos
separation = 500
vacancy = 8000
acquisition = 3000
onboard = 4000
total_direct = separation + vacancy + acquisition + onboard
total_direct

15500

In [65]:
# Productividad perdida
lost_revenue = 166000 # https://www.statista.com/statistics/934306/professional-services-worldwide-revenue-per-employee/
work_days = 240
position_vacancy_days = 40
onboarding_days_training = 60
effectiveness_during_onboarding = 0.50
lost_productivity = lost_revenue / work_days * (position_vacancy_days + onboarding_days_training * effectiveness_during_onboarding)
lost_productivity

48416.666666666664

In [67]:
# Dinero ahorrado por pérdida
salary_benefit_reduction = salary / work_days * position_vacancy_days
salary_benefit_reduction

13333.333333333332

In [68]:
# Costo de pérdida por empleado
cost_per_employee = total_direct + lost_productivity - salary_benefit_reduction
cost_per_employee

50583.33333333333

In [77]:
print("Total de empleados: {}".format(df.count()))
print("Empleados perdidos: {}".format(df.where("Termd == 1").count()))

Total de empleados: 401
Empleados perdidos: 103


In [72]:
# Costo total de pérdida de empleados
employee_number = df.count()
attrition = df.where("Termd == 1").count()
total_cost = attrition * cost_per_employee
total_cost

5210083.333333333

## El costo de perder un empleado es de 50.583 dólares al año
## Al perder 103 empleados hemos perdido 5.210.083 dólares aproximandamente